src/main/knn/KnnGraphTester.java

```diff
@@ -387,11 +409,23 @@ private void run(String... args) throws Exception {
   }
 
   private String formatIndexPath(Path docsPath) {
+    List<String> suffix = new ArrayList<>();
+    suffix.add(Integer.toString(maxConn));
+    suffix.add(Integer.toString(beamWidth));
     if (quantize) {
-      return INDEX_DIR + "/" + docsPath.getFileName() + "-" + maxConn + "-" + beamWidth + "-"
-              + quantizeBits + (quantizeCompress ? "-compressed" : "" ) + ".index";
+      suffix.add(Integer.toString(quantizeBits));
+      if (quantizeCompress == true) {
+        suffix.add("-compressed");
+      }
     }
-    return INDEX_DIR + "/" + docsPath.getFileName() + "-" + maxConn + "-" + beamWidth + ".index";
+    if (parentJoin) {
+      suffix.add("parentJoin");
+    }
+    return INDEX_DIR + "/" + docsPath.getFileName() + "-" + String.join("-", suffix) + ".index";
+  }
+
+  private boolean isParentJoinIndex(Path indexPath) {
+    return indexPath.toString().contains("parentJoin");
```

Comment:
Hey, maybe fix `KnnIndexer` to NOT allow this substring when the index is NOT a parent join index?

Reply:
KnnIndexer only adds `parentJoin` suffix to parent-join indexes. However, KnnGraphTester does allow users to provide an indexPath using `-indexPath` param. This function is an early check to prevent us from using an older, non-parent join index.
We also have a better `CheckJoinIndex(...)` validation before we actually run the parent join query later.